core/src/App.svelte

```diff
@@ -897,26 +893,51 @@
   setContext('getUnsavedChangesModalPromise', getUnsavedChangesModalPromise);
 
   //// MICRO-FRONTEND MODAL
+  // list containing the opened modals
+  let mfModalList = [];
 
-  let mfModal = {};
-
-  const resetMicrofrontendModalData = () => {
-    mfModal.displayed = false;
-    mfModal.nodepath = undefined;
-    mfModal.settings = {};
-    modalIframe = undefined;
+  /**
+   * Resets the mf modal data given the index and updates the 'mfModalList'. If no index given, resets the whole list instead
+   * @param index {number|undefined}  the index of the modal to reset
+   */
+  const resetMicrofrontendModalData = (index) => {
+    if (index !== undefined) {
+      let tempMfModalList = mfModalList;
+      // use splice to remove the modal element and re-index the list.
+      tempMfModalList.splice(index, 1);
+      // svelte components rendered from a list, expect list changes to be done through variable re-assignment
+      mfModalList = tempMfModalList;
+    } else {
+      // reset all modal list
+      mfModalList = [];
+    }
```

Comment:
```suggestion
    if (typeof index === 'undefined') {
      // reset all modal list
      mfModalList = [];
      return;
    }
    
    let tempMfModalList = mfModalList;
      // use splice to remove the modal element and re-index the list.
      tempMfModalList.splice(index, 1);
      // svelte components rendered from a list, expect list changes to be done through variable re-assignment
      mfModalList = tempMfModalList;
```
Early return is cleaner.